src/hugr/validate.rs

```diff
@@ -558,6 +521,82 @@ impl<'a, 'b> ValidationContext<'a, 'b> {
         }
         .into())
     }
+
+    fn validate_subtree(
```

Comment:
Add a comment explaining what this does, like the previous "validate op-specific constraints"

Reply:
Been a bit more specific!